README.md

```diff
@@ -1199,6 +1199,77 @@ interface Config {
 }
 ```
 
+Case of Array, you can create a read-only array by using `ReadonlyArray<T>`.
+do not allow changes such as `push()` and `fill()`, but can use features such as `concat()` and `slice()` that do not change the value.
+
+**Bad:**
+
+```ts
+const array: Array<number> = [ 1, 3, 5 ];
+array = []; // error
+array.push(100); // array will updated
+```
+
+**Good:**
+
+```ts
+const array: ReadonlyArray<number> = [ 1, 3, 5 ];
+array = []; // error
+array.push(100); // error
+```
+
+Declaring read-only arguments in [TypeScript 3.4 is a bit easier](https://github.com/microsoft/TypeScript/wiki/What's-new-in-TypeScript#improvements-for-readonlyarray-and-readonly-tuples).
+
+```ts
+function hoge(args: readonly string[]) {
+  args.push(1); // error
+}
+```
+
+Prefer [const assertions](https://github.com/microsoft/TypeScript/wiki/What's-new-in-TypeScript#const-assertions) for literal values
+
+**Bad:**
+
+```ts
+// const object
```

Comment:
The comment is redundant. Declaration of `const config = { ` already says it. I'd suggest that you remove the comment.